python/lsst/daf/butler/core/_butlerUri/http.py

```diff
@@ -345,18 +263,16 @@ def session(self) -> requests.Session:
                 refreshToken(ButlerHttpURI._session)
             return ButlerHttpURI._session
 
-        baseURL = self.scheme + "://" + self.netloc
-
-        if isWebdavEndpoint(baseURL):
-            log.debug("%s looks like a Webdav endpoint.", baseURL)
-            s = getHttpSession()
-        else:
-            raise RuntimeError(f"Only Webdav endpoints are supported; got base URL '{baseURL}'.")
-
+        s = getHttpSession()
         ButlerHttpURI._session = s
         ButlerHttpURI._sessionInitialized = True
         return s
 
+    @property
+    def is_webdav_endpoint(self) -> bool:
+        """Check if the current endpoint implements WebDAV features"""
+        return isWebdavEndpoint(self.getrooturl())
```

Comment:
Cache this in `self._is_webdav` so that we don't need the call to the server each time.
Something like:

```python
@property
def is_webdav_endpoint(self) -> bool:
    """..."""
    try:
        return self._is_webdav
    except AttributeError:
        pass
    self._is_webdav = isWebdavEndpoint(str(self.root_uri())
    return self._is_webdav
```